pkg/pillar/base/logobjecttypes.go

```diff
@@ -163,9 +163,23 @@ func InitLogObject(object *LogObject, objType LogObjectType, objName string, obj
 	}
 	object.Initialized = true
 	object.Fields = fields
+	if logBase != nil {
+		object.Merge(logBase)
+	}
```

Comment:
Similar to the other log objects, should we have a map for the source log objects also? Given the service name get me the corresponding log object?
With this map we can lookup for the log object instead of passing it as parameter to all functions. Entry functions in a particular file should do the logobject lookup and then pass around the pointer to other smaller/called functions in the same file.

Reply:
I thought about that @gkodali-zededa but it means that we'd need to pass around an agentName string so we can do the lookup. I think it is cleaner to pass around a LogObject and then just call log.Infof etc on that, as opposed to doing a lookup(agentName) at the top of each function.